job/job.go

```diff
@@ -73,8 +75,9 @@ func GetGenericJob(obj Object, config Config) (*batchv1.Job, error) {
 					RestartPolicy: corev1.RestartPolicyOnFailure,
 					Containers: []corev1.Container{
 						{
-							Name:  obj.GetMetaObject().GetName(),
-							Image: cfg.Config.BackupImage,
+							Name:      obj.GetMetaObject().GetName(),
+							Image:     cfg.Config.BackupImage,
+							Resources: getK8upResourceLimits(obj.GetResourceLimits()),
```

Comment:
This works perfectly for all the jobs. But the schedule will never reach this point. If the resource limits are defined in the schedule itself it should populate the resource limits of the job definitions it contains, or they won't be applied at all.

Just had an idea: the job definitions in the schedule are just instances of the various job spec types. So it would be possible to set a resource limit on the schedule itself but then also on the various job definitions, too. Example:

```
apiVersion: backup.appuio.ch/v1alpha1
kind: Schedule
metadata:
  name: schedule-test
spec:
  k8upResourceLimits:
    requests:
        memory: "64Mi"
        cpu: "250m"
    limits:
      memory: "128Mi"
      cpu: "500m"
  backup:
    k8upResourceLimits:
      limits:
        memory: "256Mi"
        cpu: "1000m"
    schedule: '* * * * *'
    keepJobs: 4
    promURL: http://10.144.1.224:9000
  check:
    schedule: '*/5 * * * *'
    promURL: http://10.144.1.224:9000
```

In this sample the scheduled backup should receive a merge between the limits defined in the schedule and the limits in the backup definition.

The check should simply get the value defined on the schedule itself.

Reply:
Hm, I'm a little bit unhappy with the name `k8upResourceLimits`. what about:
```yaml
apiVersion: backup.appuio.ch/v1alpha1
kind: Schedule
metadata:
  name: schedule-test
spec:
  resourceRequirementsTemplate:
    requests:
        memory: "64Mi"
        cpu: "250m"
    limits:
      memory: "128Mi"
      cpu: "500m"
  backup:
    resources:
      limits:
        memory: "256Mi"
        cpu: "1000m"
    schedule: '* * * * *'
    keepJobs: 4
    promURL: http://10.144.1.224:9000
  check:
    schedule: '*/5 * * * *'
    promURL: http://10.144.1.224:9000
```

Reply:
@Kidswiss I have a couple of questions. 
- In case of the backup schedule, how should we handle the scenario when `resourceRequirementsTemplate` is provided only in the `backup`? Do we get the default values for `schedule` and add them as well to the ones provided in the `backup`?
- Also, if `backup` and `schedule`, don't have `resourceRequirementsTemplate`, then we just add the defaults to both, right?

Reply:
My idea and understanding is:
* The `resourceRequirementsTemplate` is only in the `.spec` of `Schedule`. This serves as a template to be overridden by `.spec.backup.resources`. So, if there is no template specified, but in `.spec.backup.resources` there are values, well, there is nothing to override, just apply directly.
In other words, `.spec.backup` doesn't have the field `resourceRequirementsTemplate`.
* I think you mean: If `.spec.backup` and `.spec.check` don't have a `resource` field, then apply the values from the `.spec.resourceRequirementsTemplate` property, right?

The point of this story is to provide defaults in the template, but make it possible to override, e.g. if a certain big database backup takes 3 gigs of memory, but all the other jobs take only 100 MB.

Reply:
> .spec.check.resources would be left unspecified.

Only if the global default is not specified. If there's a global default it should be applied if nothing is defined.

So the flow should go like:

global defaults are overwritten by `resourceRequirementsTemplate` are overwritten by `resources`.

Reply:
Ok, I understand what you mean. But in the second question I meant what happens when none are provided, but i guess I just add the default ones to the, let's call them children jobs (all but schedule). Also I realized that I was thinking on adding the `resourceRequirementsTemplate` as limits to the schedule job itself, what it's probably not needed.

Reply:
@Kidswiss ok, I see. But the schedule doesn't need to have limits.